### ISSUE TITLE
Nice project, how do you do async?

### ISSUE THREAD

**Issue by jskrzypek** [mattallty/Caporal.js#3]
I can't see here how you manage async actions for commands. You have promises in the `complete()` method, can I return a promise in an action and have it handled correctly?

**Comment by mattallty** [comment]
Hi @jskrzypek, handling async actions are not possible indeed, but It should be easy to  implement though. I'm going to work on it.

**Comment by mattallty** [comment]
@jskrzypek Implemented through PR #6. In case of the Promise fails, Caporal will output an error with the rejected reason.

**Comment by ihorskyi** [comment]
@jskrzypek starting from node v7.6.0 you can use **async/await**. For example

```javascript
#!/usr/bin/env node
const Promise = require('bluebird')
const prog    = require('caporal')

prog
  .version('1.0.0')
  .action(async (args, options, logger) => {
    const result = await Promise.delay(1000).then(() => 'Some val')
    console.log(result) // outputs 'Some val' after 1 second
  })

prog.parse(process.argv)
```

**Comment by mattallty** [comment]
Implemented in #6, closing, thanks for your feedback @jskrzypek  👍
